syntax = "proto3";

option csharp_namespace = "ProfTestium_TestService";

package greet;

service TestsListSender{
	rpc GetTestsList(TestsListSenderRequest) returns(TestsListSenderReply);
}
message TestsListSenderRequest{

}

message QuestionReply{
	string QuestionText = 1;
	int32 QuestionTypeID = 2;
	int32  MaxScore = 3;
	int32 PictureID = 4;
	string PicturePath = 5;
	int32 QuestionID = 6;
}
message TestReply{
	int32 TestID = 1;
	string TestName = 2;
	repeated QuestionReply Questions = 3;
}
message TestsListSenderReply{
	repeated TestReply Tests = 1;
}
